app/models/user.rb

```diff
@@ -41,6 +41,11 @@ class User < ApplicationRecord
 
   enum gender: { male: 0, female: 1, other: 2 }
 
+  has_many :targets, dependent: :destroy
+  has_many :conversation_users, dependent: :destroy
+  has_many :conversations, through: :conversation_users
+  has_many :messages, dependent: :destroy
```

Comment:
`dependent: :destroy` is unnecessary here since you're deleting them through conversations

Reply:
RuboCop: Specify a `:dependent` option.